stan/math/prim/fun/matrix_exp_action_handler.hpp

```diff
@@ -104,26 +169,68 @@ class matrix_exp_action_handler {
    * @param [out] m int parameter m
    * @param [out] s int parameter s
    */
-  inline void set_approximation_parameter(const Eigen::MatrixXd& mat,
-                                          const double& t, int& m, int& s) {
-    if (l1norm(mat) < tol || t < tol) {
+  template <typename EigMat1, typename EigMat2,
+            require_all_eigen_t<EigMat1, EigMat2>* = nullptr,
+            require_all_st_same<double, EigMat1, EigMat2>* = nullptr>
+  inline void set_approx_order(const EigMat1& mat, const EigMat2& b,
+                               const double& t, int& m, int& s) {
+    // L1 norm
+    double normA = mat.colwise().template lpNorm<1>().maxCoeff();
+
+    if (normA < tol || t < tol) {
```

Comment:
I almost want to split the condition. If `t < tol`, then we don't have to take a norm.

I might do that.